### PR TITLE
PWN-1578 - Transaction History main feature

### PR DESCRIPTION
**Pull request by davran1001**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-1578

## Description of Work

Transaction History main feature

### CODE REVIEW

**Review by kuznetsov — APPROVED** [review]

**Review by kabramovp2p — APPROVED** [review]

**Comment by gslevinkov** [comment]
_Co-authored-by: “khasanov.davran” <“[khasanov.davran@p2p.org](mailto:khasanov.davran@p2p.org)”>_?

**Comment by kabramovp2p** [comment]
> _Co-authored-by: “khasanov.davran” <“[khasanov.davran@p2p.org](mailto:khasanov.davran@p2p.org)”>_?

yes lets get rid of this authored copy rights =)

File: app/src/main/java/org/p2p/wallet/history/ui/token/TokenHistoryFragment.kt [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+        setNavigationOnClickListener { popBackStack() }
+        if (BuildConfig.DEBUG) {
+            inflateMenu(R.menu.menu_history)
+            setOnMenuItemClickListener {
+                var isHandled = false
+                if (it.itemId == R.id.closeItem) {
+                    presenter.closeAccount()
+                    isHandled = true
+                }
+                isHandled
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/history/ui/token/TokenHistoryFragment.kt:107]: wtf? 
just return boolean directly

**Review by 1sakov — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/p2p/wallet/history/ui/token/TokenHistoryFragment.kt (hunk 1 above)
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/history/ui/token/TokenHistoryFragment.kt:107]: well that's my code. I couldn't understand when and what should this lambda return. What does it mean to return false or return true. So I created this val to make it explicit. Can be inlined, if there is need.

Comment (1sakov) [reply app/src/main/java/org/p2p/wallet/history/ui/token/TokenHistoryFragment.kt:108]: But isn't it understandable when you return it directly?
```
{
      if (someBoolean) {
            // do smth
            return@someListener true
      }
      
      return@someListener false
}
```

the boolean variable is redundant here

**Pull request merged**